### ISSUE TITLE
File manager log output

### ISSUE THREAD

**Issue by sir-hashpipe** [nelu/rutorrent-thirdparty-plugins#24]
I have an issue with Filemanager not displaying log output in the Console window.
All commands work such as rm, copy etc, it's just not displayed correctly.
I'm using rutorrent 3.8 on ubuntu 16.04. It works fine on my older machines running ubuntu 12.04 and rutorrent 3.6.
I have narrowed the problem down to that the files pid and log in folder /tmp/.rutorrent/.fman/usersession/ only gets permission 640 and owner is my own user.
Hence nginx user and rutorrent/filemanager can't read them.
On my older setups the files get permisson 644, so that also includes Read by Others.

If I manually change the files to 644 while the operation is ongoing it works fine and progress is shown in the Console log. But as these get created and removed for every task that is not a viable solution.

I was thinking about adding a chmod or setfacl to the scripts, as a workaround.

But I would really like to get to the bottom of the issue. I have looked through every file I can think of to understand why this happens, but I'm no programmer. If I make a copy of the script and mod it so I can run a test manually with some whatever echo>test etc it works and the file gets created with 644 as should be standard.

Has anyone seen a similar issue? I don't know if the issue is from OS, rutorrent, or some config.
The install is fairly new so I have not made any special modifications.

**Comment by sir-hashpipe** [comment]
Managed to get Console to work by modifying all scripts to create and set permissions on log.

Eg:

`touch "$1/log" && chmod 644 "$1/log";`

and for php sfv scripts:

`chmod($argv[1].'/log',` 0644);`

Still curious about the reason why original scripts won't work, might be something with ACL in Ubuntu 16.04?
Folders in /tmp/.rutorrent gets owned by webuser, but files by my user that is also running rtorrent.
Perhaps my user gets different umask when being run as script causing files to be created with 640?

**Comment by sir-hashpipe** [comment]
Managed to solve the issue.
I saw that all my downloaded torrents also were created with 750 for folders and 640 for files, as opposed to 755/644 on my old machines... making me suspect rtorrent as the cause.

Found following in the new default .rtorrent.rc template which I base mine on:

`system.umask.set = 0027`

Source https://github.com/rakshasa/rtorrent/wiki/CONFIG-Template

Changed this back to 0022 and all is well.
Now all scripts work fine without mods.

Hopefully this can help someone else that will encounter the same problem.
My rtorrent version is 0.96 / libtorrent 0.13.6
